src/main/java/org/opensearch/timeseries/rest/handler/AbstractTimeSeriesActionHandler.java

```diff
@@ -398,22 +407,215 @@ protected void validateTimeField(boolean indexingDryRun, ActionListener<T> liste
+                pipelineBuilder.startObject();
+                {
+                    pipelineBuilder.startObject("script");
+                    {
+                        pipelineBuilder.field("lang", "painless");
+                        String flattenScript = getScripts(TimeSeriesSettings.FLATTEN_CUSTOM_RESULT_INDEX_PAINLESS);
+                        pipelineBuilder.field("source", flattenScript);
+                    }
+                    pipelineBuilder.endObject();
+                }
+                pipelineBuilder.endObject();
+            }
+            pipelineBuilder.endArray();
+        }
+        pipelineBuilder.endObject();
+        return BytesReference.bytes(pipelineBuilder);
+    }
+
+    protected void updateResultIndexSetting(String pipelineId, String flattenedResultIndex, ActionListener<T> listener) {
+        UpdateSettingsRequest updateSettingsRequest = new UpdateSettingsRequest();
+        updateSettingsRequest.indices(flattenedResultIndex);
+
+        Settings.Builder settingsBuilder = Settings.builder();
+        settingsBuilder.put("index.default_pipeline", pipelineId);
+
+        updateSettingsRequest.settings(settingsBuilder);
+
+        client.admin().indices().updateSettings(updateSettingsRequest, ActionListener.wrap(response -> {
+            if (response.isAcknowledged()) {
+                logger.info("Successfully updated settings for index: {} with pipeline: {}", flattenedResultIndex, pipelineId);
+                listener.onResponse(null);
+            } else {
+                String errorMsg = "Settings update not acknowledged for index: " + flattenedResultIndex;
+                logger.error(errorMsg);
+                listener.onFailure(new OpenSearchStatusException(errorMsg, RestStatus.INTERNAL_SERVER_ERROR));
+            }
+        }, exception -> {
+            logger.error("Failed to update settings for index: {} with pipeline: {}", flattenedResultIndex, pipelineId, exception);
+            listener.onFailure(exception);
+        }));
+    }
+
+    private void handleFlattenResultIndexMappingUpdate(Config existingConfig, ActionListener<T> listener) {
+        if (config.getCustomResultIndexOrAlias() == null) {
+            return;
+        }
+        if (Boolean.TRUE.equals(existingConfig.getFlattenResultIndexMapping())
+            && Boolean.FALSE.equals(config.getFlattenResultIndexMapping())
+            && existingConfig.getCustomResultIndexOrAlias() != null) {
+            String pipelineId = timeSeriesIndices.getFlattenResultIndexIngestPipelineId(config.getId());
+            client.admin().cluster().deletePipeline(new DeletePipelineRequest(pipelineId), new ActionListener<AcknowledgedResponse>() {
+
+                @Override
+                public void onResponse(AcknowledgedResponse acknowledgedResponse) {
+                    if (acknowledgedResponse.isAcknowledged()) {
+                        logger.info("Ingest pipeline deleted successfully for pipelineId: {}", pipelineId);
```

Comment:
you do not seem to return listener in this branch.

Reply:
Also, do you need to remove the index.default_pipeline setting from the index?

Reply:
updated with deleting ingest pipeline and flatten result index alias